core/src/main/java/org/apache/calcite/sql/dialect/PostgresqlSqlDialect.java

```diff
@@ -113,6 +113,17 @@ public PostgresqlSqlDialect(Context context) {
         SqlParserPos.ZERO);
   }
 
+  public void quoteStringLiteral(StringBuilder buf, @Nullable String charsetName,
+      String val) {
```

Comment:
what was the need for this chnage?

Reply:
calcite unparses utf16 characters as a unicode decoded string, which is not required in the case of postgres. thats why this method is overriden. i have added a test case called `testPostgresUnicodeString` please check that it shows the difference b/w calcite dialect unparse and postgres unparse of the same string.